source/connect/tls.txt

```diff
@@ -0,0 +1,271 @@
+
+.. _c-specify-ca-file:
+
+Specify a CA File
+------------------
+
+When you connect to a MongoDB deployment with TLS enabled, the deployment will by default require the client to provide
+a client certificate issued by a certificate authority, or an authority
+trusted by the native certificate store in use on the server.
+
+You can provide the client certificate in the following ways:
+
+- Setting the ``tlscertificatekeyfile`` parameter in your connection string to a `.pem` file containing the root certificate chain
+- Using the `mongoc_uri_set_option_as_utf8()` function to set the `MONGOC_URI_TLSCERTIFICATEKEYFILE` option
+  to a `.pem` file containing the root certificate chain
+
+.. tabs::
+
+   .. tab:: Connection String
+      :tabid: connectionstring
+
+      .. literalinclude:: /includes/connect/tls.c
+         :language: c
+         :start-after: start-connect-ca-file-str
+         :end-before: end-connect-ca-file-str
+         :dedent:
+
+   .. tab:: MongoC URI Options
+      :tabid: mongocurioptions
+
+      .. literalinclude:: /includes/connect/tls.c
+         :language: c
+         :start-after: start-connect-ca-file-uri
+         :end-before: end-connect-ca-file-uri
+         :dedent:
+
+.. _c-certificate-revocation:
+
+Server Certificate Verification
+-------------------------------
+
+The MongoDB C Driver will automatically verify the validity of a server certificate issued
+by the configured Certificate Authority. The driver also performs hostname validation and revocation checking.
+
+To overwrite this behavior, it is possible to disable hostname validation, OCSP endpoint revocation checking,
+all revocation checking, and allow invalid certificates.
+
+This behavior is controlled using the ``tlsAllowInvalidHostnames``, ``tlsDisableOCSPEndpointCheck``,
+``tlsDisableCertificateRevocationCheck``, and ``tlsAllowInvalidCertificates`` options. By default,
+all are set to ``false``.
+
+It is not recommended to change these defaults, since you might expose your client to the following security risks:
+
+- `Man In The Middle attacks <https://en.wikipedia.org/wiki/Man-in-the-middle_attack>`_, when
+  ``tlsAllowInvalidHostnames`` is set
+- Invalid certificates, when ``tlsAllowInvalidCertificates`` is set
+- Potentially revoked certificates, when ``tlsDisableOCSPEndpointCheck`` or ``tlsDisableCertificateRevocationCheck`` are set
+
+Supported Libraries
+-------------------
+
+By default, libmongoc will attempt to find a supported TLS library and enable TLS support. This is controlled
+by the cmake flag ``ENABLE_SSL``, which is set to ``AUTO`` by default. This flag accepts the following values:
+
+- ``AUTO``: Links to the system's native TLS library, or attempts to find OpenSSL. This is the default value.
+- ``OPENSSL``: Links to OpenSSL (libssl). An optional install path may be specified with ``OPENSSL_ROOT``.
+- ``LIBRESSL``: Links to LibreSSL's libtls. You can link to LibreSSL's compatible libssl by setting ``OPENSSL``.
```

Comment:
```suggestion
- ``LIBRESSL`` (Deprecated):  Links to LibreSSL's libtls. You can link to LibreSSL's compatible libssl by setting ``OPENSSL``.
```

Suggest noting LIBRESSL is deprecated. [C driver 1.28.0](https://github.com/mongodb/mongo-c-driver/releases/tag/1.28.0) deprecates use of LibreSSL:

> `ENABLE_SSL=LIBRESSL` is deprecated along with `mongoc_stream_tls_libressl_new`. Support for LibreSSL may be dropped in a future major release.